### PR TITLE
Casmuser-3060: Update to lts/csm-1.3 vendor

### PR DESCRIPTION
**Pull request by alanm-hpe**
## Summary and Scope

Update the repo to use the `lts/csm-1.3` branch from node-images. This bring substantial changes to align
with the updates to node-images from `lts/csm-1.2`. 

A summary of changes include:
* Restructure Jenkinsfile.github
* Restructure application.pkr.hcl to match ncn-common.pkr.hcl
* Restructure provisioners to match ncn-common.pkr.hcl
* Use of ansible to configure the application image 
* Add cleanup.sh from ncn-node-images to remove credentials
* Add `dmsquash-live livenet` to dracut-lib.sh
* Add a systemd unit file to reduce `/proc/cmdine` permissions to 0400 before multi-user mode

## Issues and Related PRs

* Resolves [CASMUSER-3060](https://jira-pro.its.hpecorp.net:8443/browse/CASMUSER-3060)

## Testing

`rocket`

### Test description:

Verified the new image would boot. Performed the following checks
* No root password by default
* spire-agent started
* /proc/cmdline could not be read by a regular user so the presigned url could not be determined
* DHCP hostname could be set

## Risks and Mitigations

There may be unforeseen changes from `lts/csm-1.2` to `lts/csm-1.3`, however, this feature is a tech preview
and will continue to be worked on. This is an important change to pick up security fixes via vendoring of
node-images and csm-rpms. Future updates will be much easier now that this change has been made.

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [ ] CHANGELOG.md updated
- [ ] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: boxes/application/variables.pkr.hcl [hunk 1]
```diff
@@ -1,203 +1,209 @@
 variable "cpus" {
-  type = string
+  type    = string
```
Comment (keopp-hpe) [inline boxes/application/variables.pkr.hcl:2]: There seems to be extra space between 'type' and the '=' sign.  Is that intended?

File: vendor/github.com/Cray-HPE/node-images/Jenkinsfile.github [hunk 2]
```diff
@@ -25,314 +67,373 @@ pipeline {
...
+                        PKR_VAR_pit_slug = "${GIT_TAG}/${BUILD_DATE}/g${GIT_HASH}"
                     }
                     steps {
                         withCredentials([
-                            string(credentialsId: 'sles15-initial-root-password', variable: 'SLES15_INITIAL_ROOT_PASSWORD'),
-                            usernamePassword(credentialsId: 'artifactory-algol60', usernameVariable: 'ARTIFACTORY_USER', passwordVariable: 'ARTIFACTORY_TOKEN'),
-                            file(credentialsId: 'google-image-manager', variable: 'GOOGLE_APPLICATION_CREDENTIALS'),
-                            file(credentialsId: 'google-image-manager', variable: 'GOOGLE_CLOUD_SA_KEY')
+                                string(credentialsId: 'sles15-initial-root-password', variable: 'SLES15_INITIAL_ROOT_PASSWORD'),
+                                usernamePassword(credentialsId: 'artifactory-algol60', usernameVariable: 'ARTIFACTORY_USER', passwordVariable: 'ARTIFACTORY_TOKEN'),
```
Comment (keopp-hpe) [inline vendor/github.com/Cray-HPE/node-images/Jenkinsfile.github:232]: Extra whitespace in several lines here.

File: vendor/github.com/Cray-HPE/node-images/Jenkinsfile.github [hunk 3]
```diff
@@ -25,314 +67,373 @@ pipeline {
...
-                        expression { params.buildGoogle }
-                    }
+                stage('Storage-CEPH') {
                     steps {
                         withCredentials([
-                            string(credentialsId: 'sles15-initial-root-password', variable: 'SLES15_INITIAL_ROOT_PASSWORD'),
-                            usernamePassword(credentialsId: 'artifactory-algol60', usernameVariable: 'ARTIFACTORY_USER', passwordVariable: 'ARTIFACTORY_TOKEN'),
-                            file(credentialsId: 'google-image-manager', variable: 'GOOGLE_APPLICATION_CREDENTIALS'),
-                            file(credentialsId: 'google-image-manager', variable: 'GOOGLE_CLOUD_SA_KEY')
+                                string(credentialsId: 'sles15-initial-root-password', variable: 'SLES15_INITIAL_ROOT_PASSWORD'),
```
Comment (keopp-hpe) [inline vendor/github.com/Cray-HPE/node-images/Jenkinsfile.github:328]: Extra whitespace here, too.

File: boxes/application/variables.pkr.hcl (hunk 1 above)
Comment (alanm-hpe) [reply boxes/application/variables.pkr.hcl:2]: Intended. It aligns with the following line. All of the spacing and ordering changes are meant to align with common.pkr.hcl so in it's easier to diff when we vendor node-images again.

**Review by keopp-hpe — APPROVED** [review]

**Pull request merged**